File: medium/python/554.py
# https://leetcode.com/problems/brick-wall/

class Solution:
    def leastBricks(self, wall: List[List[int]]) -> int:       
        counts = collections.defaultdict(int)
        
        for row in wall:
            pos = 0
            
            for i in row[:-1]:
                pos += i
                counts[pos] += 1
                
        values = counts.values()
                
        return len(wall) - (0 if len(values) == 0 else max(counts.values()))

File: medium/cpp/554.cpp
// https://leetcode.com/problems/brick-wall/

class Solution
{
public:
    int leastBricks(std::vector<std::vector<int>>& wall)
    {
        const auto n = std::size(wall);
        
        std::unordered_map<int, int> counts;
        
        for (const auto& row : wall)
        {
            auto pos = 0;
            
            for (auto i = 0; i + 1 < std::size(row); i++)
            {
                pos += row[i];
                counts[pos]++;
            }
        }
        
        auto ans = n;
        
        for (const auto& entry : counts)
        {
            const auto pos = entry.first;
            const auto edges = entry.second;
            
            ans = std::min(ans, n - edges);
        }
        
        return ans;
    }
};